rct229/rules/section6/section6rule7.py

```diff
@@ -0,0 +1,107 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+
+MSG_WARN_DAYLIGHT_NO_SCHEDULE = "Some of the spaces in zone are modeled with window(s) and/or skylight(s) and have daylighting controls modeled explicitly in the simulation tool. Verify that the mandatory lighting control requirements are met."
+MSG_WARN_DAYLIGHT = "Some of the spaces in zone are modeled with window(s) and/or skylight(s) and have daylighting controls modeled via schedule adjustment. Verify that the mandatory lighting control requirements are met, and that the supporting documentation is provided for the schedule adjustment."
+MSG_WARN_NO_DAYLIGHT = "Some of the spaces in zone are modeled with fenestration but no daylighting controls. The design must include mandatory daylighting controls unless any of the exceptions to 90.1 section 9.4.1.1€ apply."
+
+
+class Section6Rule7(RuleDefinitionListIndexedBase):
+    """Rule 7 of ASHRAE 90.1-2019 Appendix G Section 6 (Lighting)"""
+
+    def __init__(self):
+        super(Section6Rule7, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, False, True),
+            each_rule=Section6Rule7.ZoneRule(),
+            index_rmr="proposed",
+            id="6-7",
+            description="Proposed building is modeled with daylighting controls directly or through schedule adjustments.",
+            list_path="ruleset_model_instances[0].buildings[*].building_segments[*].zones[*]",
+        )
+
+    class ZoneRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section6Rule7.ZoneRule, self,).__init__(
+                required_fields={"$": ["spaces", "surfaces"]},
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+            )
+
+        def get_calc_vals(self, context, data=None):
+            zone_p = context.proposed
+
+            daylight_flag_p = (
+                len(
+                    find_all(
+                        "$..surfaces[?adjacent_to = 'EXTERIOR'].subsurfaces[?classification != 'DOOR']",
```

Comment:
I will prefer to use enum instead of string here. See [5-24](https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/rules/section5/section5rule24.py) as an example.
You can replace both `adjacent_to` and `classification' to the enum.

Reply:
Great point! Fixed.

Reply:
Great point! Fixed.